### PR TITLE
Folder structure changes and renaming

### PR DESCRIPTION
**Pull request by zhenwendai**
It contains the following changes:
- Folder structure changes and renaming according to our discussion.
- Extend the Mark's multivariate normal distribution implementation to allow general broadcasting.

@marpulli I changed the behavior of your log_pdf and kl_divergence implementation to return an array of the size of mean without the last dimension. This is consistent with our other distributions.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: mxfusion/runtime/distributions/distribution.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # ==============================================================================
 
 
-class DistributionImplementation(object):
+class RuntimDistribution(object):
```
Comment (meissnereric) [inline mxfusion/runtime/distributions/distribution.py]: RunTimDistribution is misspelled (RunTim -> RunTime)

File: mxfusion/runtime/distributions/multivariate_normal.py [hunk 2]
```diff
@@ -12,38 +12,50 @@
...
         :param mean: Mean of the normal distribution. Shape: (n_samples, n_outputs, n_dim)
         :type mean: MXNet NDArray
         :param covariance: Covariance matrix of the distribution. Shape: (n_samples, n_outputs, n_dim, n_dim)
         :type covariance: MXNet NDArray
         """
+        super(MultivariateNormal, self).__init__()
 
-        if mean.ndim != 3:
-            raise ValueError('Mean should have 3 dimensions. It has {}'.format(mean.ndim))
+        # if mean.ndim != 3:
```
Comment (meissnereric) [inline mxfusion/runtime/distributions/multivariate_normal.py]: Either remove these commented out lines or add them back in please.

**Comment by marpulli** [comment]
At the minute there is
`mxfusion.components.distributions.bernoulli.Bernoulli` and 
`mxfusion.runtime.distributions.bernoulli.Bernoulli`

It might be confusing to have two different classes with the same name in the same library.

Can we do something like `mxfusion.runtime.distributions.bernoulli.BernoulliRuntime` or 
`mxfusion.components.distributions.BernoulliFactor` to distinguish them?

**Comment by zhenwendai** [comment]
- I have renamed all the run time distributions as XXXRunTime to distinguish from the model definition distributions and leave the name of original distributions unchanged to keep interface consistency.

- I implemented the mechanism that, if the run time distribution of a model definition distribution is specified, it uses the run time distribution for the log_pdf and draw_samples computation.

**Pull request merged**